src/room/useRoomState.ts

```diff
@@ -5,25 +5,33 @@ SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 Please see LICENSE in the repository root for full details.
 */
 
-import { useCallback, useMemo, useState } from "react";
-import { type RoomState, RoomStateEvent, type Room } from "matrix-js-sdk";
+import { useCallback } from "react";
+import {
+  type RoomState,
+  RoomStateEvent,
+  type Room,
+  RoomEvent,
+} from "matrix-js-sdk";
 
-import { useTypedEventEmitter } from "../useEvents";
+import { useTypedEventEmitterState } from "../useEvents";
 
 /**
  * A React hook for values computed from room state.
  * @param room The room.
  * @param f A mapping from the current room state to the computed value.
  * @returns The computed value.
  */
-export const useRoomState = <T>(room: Room, f: (state: RoomState) => T): T => {
-  const [numUpdates, setNumUpdates] = useState(0);
-  useTypedEventEmitter(
+export function useRoomState<T>(room: Room, f: (state: RoomState) => T): T {
+  // TODO: matrix-js-sdk says that Room.currentState is deprecated, but it's not
+  // clear how to reactively track the current state of the room without it
+  const currentState = useTypedEventEmitterState(
     room,
+    RoomEvent.CurrentStateUpdated,
+    useCallback(() => room.currentState, [room]),
+  );
+  return useTypedEventEmitterState(
+    currentState,
     RoomStateEvent.Update,
-    useCallback(() => setNumUpdates((n) => n + 1), [setNumUpdates]),
+    useCallback(() => f(currentState), [f, currentState]),
   );
```

Comment:
why id is it not enough, to listen to `RoomEvent.CurrentStateUpdated`?
Shouldnt that also update the currentState if currentState emits RoomStateUpdate.Update

Reply:
No, it's only called when the `currentState` object is replaced with something else: https://github.com/matrix-org/matrix-js-sdk/blob/44399f601733c8f4daf5a6c891a355cf425192a9/src/models/room.ts#L1319